Mono.Addins/Mono.Addins/AddinLocalizerAttribute.cs

```diff
@@ -0,0 +1,66 @@
+﻿//
+// AddinLocalizerAttribute.cs
+//
+// Author:
+//       Matt Ward <matt.ward@xamarin.com>
+//
+// Copyright (c) 2017 Xamarin Inc. (http://xamarin.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+using System;
+
+namespace Mono.Addins
+{
+	/// <summary>
+	/// Declares a custom localizer for an add-in.
+	/// </summary>
+	[AttributeUsage (AttributeTargets.Assembly)]
+	public class AddinLocalizerAttribute: Attribute
+	{
+		string type;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="Mono.Addins.AddinLocalizerAttribute"/> class.
+		/// </summary>
+		public AddinLocalizerAttribute ()
+		{
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="Mono.Addins.AddinLocalizerAttribute"/> class.
+		/// </summary>
+		/// <param name='type'>
+		/// The type name of the localizer. This type must implement the
+		/// <see cref="Mono.Addins.Localization.IAddinLocalizerFactory"/> interface.
+		/// </param>
+		public AddinLocalizerAttribute (string type)
+		{
+			this.type = type;
+		}
+
+		/// <summary>
+		/// Type name of the localizer.
+		/// </summary>
+		public string Type {
```

Comment:
This should return a Type. Also, there needs to be a TypeName property that gets/sets the full name of the type. See https://github.com/mono/mono-addins/blob/master/Mono.Addins/Mono.Addins/ExtensionAttribute.cs#L156 for example.